src/libraries/Microsoft.PowerFx.Core/Functions/TexlFunction.cs

```diff
@@ -445,12 +452,19 @@ private bool CheckTypesCore(TexlNode[] args, DType[] argTypes, IErrorContainer e
         {
             Contracts.AssertValue(args);
             Contracts.AssertAllValues(args);
-            Contracts.AssertValue(argTypes);
-            Contracts.AssertAllValid(argTypes);
+            Contracts.AssertValue(argTypes);            
             Contracts.Assert(args.Length == argTypes.Length);
             Contracts.AssertValue(errors);
             Contracts.Assert(MinArity <= args.Length && args.Length <= MaxArity);
 
+            for (int i = 0; i < argTypes.Length; i++)
+            {
+                if (!IsIdentifierParam(i))
+                {
+                    Contracts.AssertValid(argTypes[i]);
```

Comment:
Did we not determine what the type of these args would be?

Reply:
Am replacing AssertAllValid with this code (line 449 in previous code)

Reply:
but now that argType is no longer null ... can we revert this change?

Reply:
This wouldn't work as DType.Invalid will make Contracts.Assert fail

Reply:
discussed offline - Identifier Params are DKind.Invalid, which would still fail the assert.